pkg/pipelines/git/repository_test.go

```diff
@@ -151,3 +152,31 @@ func TestCreateWebHook(t *testing.T) {
 		t.Errorf("failed to create webhook, got %q, want %q", created, "1")
 	}
 }
+
+func TestExtractRepo(t *testing.T) {
+	Tests := []struct {
+		desc     string
+		argument string
+		want     string
+	}{
+		{"Standard url",
+			"https://github.com/username/repo.git",
+			`username/repo`},
+		{"Standard url",
+			"https://github.com/username/repo/.git",
+			`username/repo`},
```

Comment:
How about adding a test case for the case you are fixing?
"https://github.com/username/repo/"